modules/likes/jetpack-likes-settings.php

```diff
@@ -0,0 +1,645 @@
+			}
+		}
+
+		// Record a change in like status for this post - only if it contradicts the
+		// site like setting.
+		if ( ( $this->is_enabled_sitewide() && empty( $_POST['wpl_enable_post_likes'] ) ) || ( ! $this->is_enabled_sitewide() && ! empty( $_POST['wpl_enable_post_likes'] ) ) ) {
+			update_post_meta( $post_id, 'switch_like_status', 1 );
+		} else {
+			delete_post_meta( $post_id, 'switch_like_status' );
+		}
+
+		return $post_id;
+	}
+
+	/**
+	 * Adds the 'sharing' menu to the settings menu.
+	 * Only ran if sharedaddy and publicize are not already active.
+	 */
+	function sharing_menu() {
+		add_submenu_page( 'options-general.php', esc_html__( 'Sharing Settings', 'jetpack' ), esc_html__( 'Sharing', 'jetpack' ), 'manage_options', 'sharing', array( $this, 'sharing_page' ) );
+	}
+
+	/**
+	 * Provides a sharing page with the sharing_global_options hook
+	 * so we can display the setting.
+	 * Only ran if sharedaddy and publicize are not already active.
+	 */
+	function sharing_page() {
+		$this->updated_message(); ?>
+		<div class="wrap">
+			<div class="icon32" id="icon-options-general"><br /></div>
+			<h1><?php esc_html_e( 'Sharing Settings', 'jetpack' ); ?></h1>
+			<?php
+			/** This action is documented in modules/sharedaddy/sharing.php */
+			do_action( 'pre_admin_screen_sharing' );
+			?>
+			<?php $this->sharing_block(); ?>
+		</div> <?php
+	}
+
+	/**
+	 * Returns the settings have been saved message.
+	 */
+	function updated_message() {
+		if ( isset( $_GET['update'] ) && $_GET['update'] == 'saved' )
+			echo '<div class="updated"><p>' . esc_html__( 'Settings have been saved', 'jetpack' ) . '</p></div>';
+	}
+
+	/**
+	 * Returns just the "sharing buttons" w/ like option block, so it can be inserted into different sharing page contexts
+	 */
+	function sharing_block() { ?>
+		<h2><?php esc_html_e( 'Sharing Buttons', 'jetpack' ); ?></h2>
+		<form method="post" action="">
+			<table class="form-table">
+				<tbody>
+				<?php
+				/** This action is documented in modules/sharedaddy/sharing.php */
+				do_action( 'sharing_global_options' );
+				?>
+				</tbody>
+			</table>
+
+			<p class="submit">
+				<input type="submit" name="submit" class="button-primary" value="<?php esc_attr_e( 'Save Changes', 'jetpack' ); ?>" />
+			</p>
+
+			<input type="hidden" name="_wpnonce" value="<?php echo wp_create_nonce( 'sharing-options' );?>" />
+		</form> <?php
+	}
+
+	/**
+	 * Are likes enabled for this post?
+	 *
+	 * @param int $post_id
+	 * @return bool
+	 */
+	function is_post_likeable( $post_id = 0 ) {
+		$post = get_post( $post_id );
+		if ( !$post || is_wp_error( $post ) ) {
+			return false;
+		}
+
+		$sitewide_likes_enabled = (bool) $this->is_enabled_sitewide();
+		$post_likes_switched    = (bool) get_post_meta( $post->ID, 'switch_like_status', true );
+
+		$post_likes_enabled = $sitewide_likes_enabled;
+		if ( $post_likes_switched ) {
+			$post_likes_enabled = ! $post_likes_enabled;
+		}
+
+		return $post_likes_enabled;
```

Comment:
looks like we could do some logical reduction here too if we wanted. this is a lot of variable assignment that just gets abandoned

```php
$post_likes_enabled = (bool) $this->is_enabled_sitewide();
$post_likes_switched = …

return $post_likes_switched
	? ! $post_likes_enabled
	: $post_likes_enabled;

// but of course, we can recognize this as a logical XOR

return $post_likes_switched xor $post_likes_enabled;
```